### DISCUSSION TITLE
Area weighting and voronoi tessellation

### DISCUSSION THREAD

**Discussion by rgrumbine** [dtcenter/METplus#2315] [Statistical Computation] [answered]
It seems that when matching points, such as surface stations, against gridded fields, say temperature, METplus weights the collection by number of points in computing summary statistics. This translates to 3000 mesonet stations in a compact area being weighted 100x the 30 stations that represent 100x the area.

What I would rather see is weighting by area for the matchups, assigning each grid cell to the point observation nearest to it and weight the error by the area of that grid cell. 

Partitioning a domain in to zones (finding bounding polygons of the zones) which are closest to each of N input points is a known problem mathematically and is solved by Voronoi tessellation.

I could not find it in the Gnu Scientific Library, but it is part of the python scipy module. As we discussed, it's undesirable to add new dependencies, but it's hard to imagine that METplus isn't already using scipy.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@rgrumbine, thanks for sending along these details. I decided to write up an issue to capture this feature request in MET. Please see dtcenter/MET#2661. I included some implementation questions in the issue description. And I tentatively assigned it to the MET 12.0.0 release to keep it on our radar, but the timeline really depends on how/when the details shake out.

I'd recommend that we move the discussion about that feature to the comments of that issue.

Thanks!
John Halley Gotway

**Comment by rgrumbine** [comment]
Thanks John. The issue write up looks good. I'll add a detail or two over there.
